src/main/java/com/google/api/codegen/transformer/DiscoGapicInterfaceContext.java

```diff
@@ -0,0 +1,185 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.config.DiscoGapicMethodConfig;
+import com.google.api.codegen.config.GapicProductConfig;
+import com.google.api.codegen.config.InterfaceConfig;
+import com.google.api.codegen.config.MethodConfig;
+import com.google.api.codegen.config.VisibilityConfig;
+import com.google.api.codegen.discogapic.transformer.DiscoGapicNamer;
+import com.google.api.codegen.discovery.Document;
+import com.google.api.codegen.discovery.Method;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * The context for transforming a Discovery Doc API into a view model to use for client library
+ * generation.
+ */
+@AutoValue
+public abstract class DiscoGapicInterfaceContext implements InterfaceContext {
+  public static DiscoGapicInterfaceContext createWithoutInterface(
+      Document document,
+      GapicProductConfig productConfig,
+      SchemaTypeTable typeTable,
+      DiscoGapicNamer discoGapicNamer,
+      SurfaceNamer surfaceNamer,
+      FeatureConfig featureConfig) {
+    return new AutoValue_DiscoGapicInterfaceContext(
+        document,
+        productConfig,
+        typeTable,
+        discoGapicNamer,
+        (new ImmutableList.Builder<Method>()).build(),
+        "",
+        surfaceNamer,
+        featureConfig);
+  }
+
+  public static DiscoGapicInterfaceContext createWithInterface(
+      Document document,
+      String interfaceName,
+      GapicProductConfig productConfig,
+      SchemaTypeTable typeTable,
+      DiscoGapicNamer discoGapicNamer,
+      SurfaceNamer surfaceNamer,
+      FeatureConfig featureConfig) {
+    ImmutableList.Builder<Method> interfaceMethods = new ImmutableList.Builder<>();
+
+    for (MethodConfig method : productConfig.getInterfaceConfig(interfaceName).getMethodConfigs()) {
+      interfaceMethods.add(((DiscoGapicMethodConfig) method).getMethod());
+    }
+
+    return new AutoValue_DiscoGapicInterfaceContext(
+        document,
+        productConfig,
+        typeTable,
+        discoGapicNamer,
+        interfaceMethods.build(),
+        interfaceName,
+        surfaceNamer,
+        featureConfig);
+  }
+
+  public abstract Document getDocument();
+
+  @Override
+  public abstract GapicProductConfig getProductConfig();
+
+  public abstract SchemaTypeTable getSchemaTypeTable();
+
+  public abstract DiscoGapicNamer getDiscoGapicNamer();
```

Comment:
Leaned out DiscoGapicNamer. There are a few methods that take in a sole String as parameter, and I don't want them accidentally used for Gapic models.